packages/compass-assistant/src/compass-assistant-provider.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { type PropsWithChildren, useRef } from 'react';
+import { type UIMessage } from './@ai-sdk/react/use-chat';
+import { Chat } from './@ai-sdk/react/chat-react';
+import { usePreference } from 'compass-preferences-model/provider';
+import { createContext, useContext } from 'react';
+import { registerCompassPlugin } from '@mongodb-js/compass-app-registry';
+import { atlasServiceLocator } from '@mongodb-js/atlas-service/provider';
+import { DocsProviderTransport } from './docs-provider-transport';
+
+export const ASSISTANT_DRAWER_ID = 'compass-assistant-drawer';
+
+interface AssistantContextType {
+  chat: Chat<UIMessage>;
+}
+
+export const AssistantContext = createContext<AssistantContextType | null>(
+  null
+);
+
+type AssistantActionsContextType = unknown;
+export const AssistantActionsContext =
+  createContext<AssistantActionsContextType>({});
+
+export function useAssistantActions(): AssistantActionsContextType {
+  return useContext(AssistantActionsContext);
+}
+
+export const AssistantProvider: React.FunctionComponent<
+  PropsWithChildren<{
+    chat: Chat<UIMessage>;
+  }>
+> = ({ chat, children }) => {
+  const enableAIAssistant = usePreference('enableAIAssistant');
+
+  const assistantContext = useRef<AssistantContextType>({
+    chat,
+  });
+  assistantContext.current = {
+    chat,
+  };
+
+  const assistantActionsContext = useRef<AssistantActionsContextType>({});
+
+  if (!enableAIAssistant) {
+    return <>{children}</>;
+  }
```

Comment:
Suuuuuuper nit, but I kinda feel like it's important to disable this in the drawer, but probably doesn't really hurt if we just unconditionally render providers here